tests/e2e/vsphere.go

```diff
@@ -1252,3 +1260,42 @@ func (vs *vSphere) reconfigPolicy(ctx context.Context, volumeID string, profileI
 	framework.Logf("reconfigpolicy on volume %v with policy %v is successful", volumeID, profileID)
 	return nil
 }
+
+// cnsRelocateVolumeInParallel relocates volume in parallel from one datastore to another
+// using CNS API
+func cnsRelocateVolumeInParallel(e2eVSphere vSphere, ctx context.Context, fcdID string,
+	dsRefDest vim25types.ManagedObjectReference, waitForRelocateTaskToComplete bool,
+	wg *sync.WaitGroup) {
+	defer wg.Done()
+	_, err := e2eVSphere.cnsRelocateVolume(e2eVSphere, ctx, fcdID, dsRefDest, waitForRelocateTaskToComplete)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+}
+
+// waitForCNSTaskToComplete wait for CNS task to complete
+// and gets the result and checks if any fault has occurred
+func waitForCNSTaskToComplete(ctx context.Context, task *object.Task, msg string) {
+	var pandoraSyncWaitTime int
+	var err error
+	if os.Getenv(envPandoraSyncWaitTime) != "" {
+		pandoraSyncWaitTime, err = strconv.Atoi(os.Getenv(envPandoraSyncWaitTime))
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	} else {
+		pandoraSyncWaitTime = defaultPandoraSyncWaitTime
+	}
+
+	taskInfo, err := task.WaitForResult(ctx, nil)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	taskResult, err := cns.GetTaskResult(ctx, taskInfo)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	framework.Logf("Sleeping for %v seconds to allow CNS to sync with pandora", pandoraSyncWaitTime)
+	time.Sleep(time.Duration(pandoraSyncWaitTime) * time.Second)
+
+	cnsTaskRes := taskResult.GetCnsVolumeOperationResult()
+
+	if cnsTaskRes.Fault != nil {
+		err = fmt.Errorf("failed to %s=%+v", msg, cnsTaskRes.Fault)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	}
+}
```

Comment:
Can you return cnsTaskRes.Fault from the function and let the test handle the failure occurred or not from the test

With this approach, you can use the same function for the negative scenario also

Reply:
Done